src/main/kotlin/com/cjbooms/fabrikt/generators/client/ClientGeneratorUtils.kt

```diff
@@ -37,10 +38,12 @@ object ClientGeneratorUtils {
                 }
 
         return "ApiResponse".toClassName(packages.client)
-            .parameterizedBy(returnType.copy(nullable = true))
+            .parameterizedBy(returnType.copy(nullable = isNullableReturnType()))
     }
 
     fun simpleClientName(resourceName: String) = "$resourceName${ClientType.SIMPLE_CLIENT_SUFFIX}"
 
     fun enhancedClientName(resourceName: String) = "$resourceName${ClientType.ENHANCED_CLIENT_SUFFIX}"
+
+    private fun Operation.isNullableReturnType(): Boolean = responses.all { it.key == "200" }.not()
```

Comment:
This seems a bit brittle. Is there nothing in theAPI spec that gives an indication whether the response should be nullable or not?

Reply:
Not sure 😬  I guess the right approach would be to always to rely on the `content` attribute of the response. If present then we may always treat it as `non-nullable`, `nullable` otherwise.

According to the spec:
> Some responses, such as 204 No Content, have no body. To indicate the response body is empty, do not specify a content for the response

The thing is that if the `content` attribute isn't specified then we can't know the return type, so making it nullable or non-nullable is kind of irrelevant.

Not sure what to do here. Perhaps we should always treat them as non-nullable? 🤔

Reply:
Cases like the one below look "semantically" wrong, but still "syntactically" correct. I guess the question in here is if there could exist any scenario where the response body could be optional?

```
      responses:
        201:
          description: "successful operation"
          headers:
            Location:
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/NullableObject'
```

Reply:
Yes, in that case I guess `NullableObject` is an object containing nullable properties. It itself is non nullable.

> Some responses, such as 204 No Content, have no body. To indicate the response body is empty, do not specify a content for the response

Do we currently have any use for nullable?  I'm thinking the following:
no content -> `ApiResponse<Unit>`
single content schema: MyTypeOne -> `ApiResponse<MyTypeOne>`
multiple content schemas:  -> `ApiResponse<JsonNode>` or `ApiResponse<Any>`

So I think making all response types non-nullable is the right thing to do.

Reply:
That makes perfect sense ☝️ 

> Do we currently have any use for nullable?

None on the top of my head right now except for Streaming APIs upon empty streams, which are not the case. I can't think of anything else

This `non-nullable` approach is much more consistent and beautiful 😍 

PR update coming.... 👍